File: Blog/src/Blog.Dtos/Post/PostModel.cs
using Blog.Core.Interfaces;

namespace Blog.Dtos.Post;

public class PostModel : IDto {
    public long Id { get; set; }
    public string Photo { get; set; } = string.Empty;
    public string Header { get; set; } = string.Empty;
    public string Body { get; set; } = string.Empty;
    public int Type { get; set; }

    public long CreateTime { get; set; }
    public long? UserId { get; set; }
}

File: Blog/ui/src/app/models/post.model.ts
import { PostTypeEnum } from '@shared/enums/post-type.enum';

export class PostModel {
    public id!: number;
    public photo!: string;
    public header!: string;
    public body!: string;
    public createTime!: number;
    public type!: PostTypeEnum;
    public userId?: number;

    public constructor(
        fields?: Partial<PostModel>) {

        if (fields) {

            Object.assign(this, fields);
        }
    }
} 
